### PR TITLE
Remove a cosmetic/semantic base class in the basic checker nebula

### PR DESCRIPTION
**Pull request by Pierre-Sassoulas**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |

## Description

See https://github.com/PyCQA/pylint/issues/6060#issuecomment-1085648352

### CODE REVIEW

File: pylint/checkers/base/name_checker/checker.py [hunk 1]
```diff
@@ -24,6 +23,8 @@
 from pylint.checkers.utils import is_property_deleter, is_property_setter
 
 # Default patterns for name types that do not have styles
+from pylint.interfaces import IAstroidChecker
```
Comment (DanielNoord) [inline pylint/checkers/base/name_checker/checker.py:26]: This seems incorrectly placed.

**Comment by Pierre-Sassoulas** [comment]
Closing as it was just done to check if the inheritance from the same base class was required for it to work.

**Comment by DanielNoord** [comment]
Don't we want to go ahead with this?

**Comment by Pierre-Sassoulas** [comment]
I don't know if it's worth it, there's an explicitness in inheriting from the same base class that makes it easy to see those 6 checkers are linked and the name is important. I'm neutral on the change, so adding a commit to the repository for this seems like noise. If you think we should do it then let's do it :)

**Comment by DanielNoord** [comment]
Fine to keep as is. We need to something about that `__implements__` anyway/

**Pull request closed without merging**